src/common/src/hash/key.rs

```diff
@@ -443,12 +456,12 @@ impl<'a> HashKeySerDe<'a> for StructRef<'a> {
     type S = Vec<u8>;
 
     /// This should never be called
-    fn serialize(self) -> Self::S {
+    fn fixed_size_serialize(self) -> Self::S {
         todo!()
```

Comment:
Maybe out of the scope of this PR, but this code is super anti-pattern 😇  As it said "this should never be called", then `StructRef` should not implement `HashKeySerDe`.

May need further investigation on how to refactor

Reply:
What's more, the trait bound `D: HashKeySerDe` in `HashKeySerializer::apend` and `HashKeyDeserializer::deserialize` feels as if `HashKeySerDe` is only encoding that should be used, hence the comment on `HashKeySerDe`.

Reply:
Let's open another issue after this PR being merged